inc/azure_c_util/interlocked_if.h

```diff
@@ -0,0 +1,156 @@
+#ifndef INTERLOCKEDIFIF_H
+#define INTERLOCKEDIFIF_H
+
+#include <stdint.h>
+
+long InterlockedIFAdd( 
```

Comment:
>InterlockedIFAdd [](start = 5, length = 16)

usually our API names are all_lower_case_letters_with_underline_between_words.

Reply:
I'm trying to follow the Interlocked API convention so that future work developers using this interface in the future feel like they are using the Interlocked API directly.